File: 1.Domain/Domain.Model/Process.cs
using Core;
using Domain.Model.Enum;
using System;
using System.Collections.Generic;

namespace Domain.Model
{
    public class Process : Entity<int>
    {
        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }
      
        public ProcessStatus Status { get; set; }
        public ProcessCurrentStage CurrentStage { get; set; }

        public string RejectionReason { get; set; }

        public int? CandidateId { get; set; }
        public Candidate Candidate { get; set; }

        public int? ConsultantOwnerId { get; set; }
        public Consultant ConsultantOwner { get; set; }

        public int? ConsultantDelegateId { get; set; }
        public Consultant ConsultantDelegate { get; set; }
  
        public float ActualSalary { get { return HrStage.ActualSalary; } }
        public float WantedSalary { get { return HrStage.WantedSalary; } }
        public float AgreedSalary { get { return OfferStage.AgreedSalary; } }
        public EnglishLevel EnglishLevel { get { return HrStage.EnglishLevel; } }
        public Seniority Seniority { get {
                return (OfferStage.Status != StageStatus.NA ? OfferStage.Seniority : TechnicalStage.Seniority);
            }
        }

        public DateTime OfferDate { get { return OfferStage.OfferDate; } }
        public DateTime HireDate { get { return OfferStage.HireDate; } }

        //public ICollection<Stage> Stages { get; set; }

        public HrStage HrStage { get; set; }
        public TechnicalStage TechnicalStage { get; set; }
        public ClientStage ClientStage { get; set; }
        public OfferStage OfferStage { get; set; }
    }
}

File: 4.Clients/WebApi/ClientApp/src/entities/process.ts
import { Stage } from './stage';
import { Consultant } from './consultant';
import { Candidate } from './candidate';
import { ProcessStatusEnum } from './enums/process-status.enum';
import { SeniorityEnum } from './enums/seniority.enum';
import { HrStage } from './hr-stage';
import { TechnicalStage } from './technical-stage';
import { ClientStage } from './client-stage';
import { OfferStage } from './offer-stage';
import { EnglishLevelEnum } from './enums/english-level.enum';
import { ProcessCurrentStageEnum } from './enums/process-current-stage';
import { DeclineReason } from './declineReason';

export class Process {
    id: number;
    startDate: Date;
    endDate: Date;
    status: ProcessStatusEnum;
    currentStage: ProcessCurrentStageEnum;
    candidateId: number;
    candidate: Candidate;
    consultantOwnerId: number;
    consultantOwner: Consultant;
    consultantDelegateId: number;
    consultantDelegate: Consultant;
    rejectionReason: string;
    declineReason: DeclineReason;
    actualSalary: number;
    wantedSalary: number;
    agreedSalary: number;
    englishLevel: EnglishLevelEnum;
    seniority: SeniorityEnum;
    hrStage: HrStage;
    technicalStage: TechnicalStage;
    clientStage: ClientStage;
    offerStage: OfferStage;
}
